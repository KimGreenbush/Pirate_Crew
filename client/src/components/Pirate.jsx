import React from 'react'
import DeleteButton from "../components/DeleteButton"
import {Link} from "@reach/router"

const Pirate = ({ pirate, removePirate }) => {

    return (
        <div className="pirate">
            {pirate ?
                <div>
                    <Link to={`/pirate/${pirate._id}`}><img src={pirate.image} alt="pirate" /></Link>
                    <h3>{pirate.name}</h3>
                    <p>
                    <Link to={`/pirate/${pirate._id}`}><button>View Pirate</button></Link>
                    <DeleteButton pirateId={pirate._id} successCallback={() => removePirate(pirate._id)} />
                    </p>
                </div>
                : ""}
        </div>
    )
}

export default Pirate