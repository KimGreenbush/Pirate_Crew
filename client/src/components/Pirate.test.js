/**
 * @jest-environment jsdom
 */

import React from "react";
import Pirate from "./Pirate";
import renderer from "react-test-renderer";

test("renders Pirate without crashing", () => {
	const component = renderer.create(<Pirate />);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
