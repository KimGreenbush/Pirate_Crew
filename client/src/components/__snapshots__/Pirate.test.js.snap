// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Pirate without crashing 1`] = `
<div
  className="pirate"
>
  
</div>
`;
